source/dub/dependency.d

```diff
@@ -303,6 +330,7 @@ struct Dependency {
 			json = Json.emptyObject;
```

Comment:
I don't know where toJson is called right now, however this else block with your new repository value will not be used because repository is used and thus `path` is empty and most likely non-optional, which makes it go into the other part of this if.

(I think this might be a SDL <-> JSON conversion thing which would not work here?)

Reply:
Thanks. Fixed and added a test.